### DISCUSSION TITLE
say... how accurate is the exposure detection?

### DISCUSSION THREAD

**Discussion by rebeltaz** [sn4k3/UVtools#624] [Q&A] [answered]
So, I just got a Creality Halot Lite and I cannot run the slicer (Halot Box) because I run linux and it is not stable under wine. Anyway, I loaded one of the demo models on the flash drive that came with the printer into uvtools and for the exposure settings it said: 

Bottom Layers: 8/0.4mm
Exposure: 60/0.6s
Lift: 5/50mm
Retract: 5/5mm
Wait Time: 2/0/0s

When I loaded the same model in the printer, though, and clicked on settings, IT said:

Initial Layer: 40s
Exposure: 3s
Motor Speed: 3mm/s
Turn Off Delay: 4s
Rising Height: 8mm
Bottom Layer: 1

I counted out the exposure time and what I came back with (counting one Mississippi after the plate dropped) was more like 5 seconds or so. 

I cannot find anything relating to the print settings of this printer, so I was hoping that I could use the settings detected by this program. I don't think that the exposure setting is 0.6 seconds, though. How accurate is the detection of those settings?

**Comment by sn4k3** [comment, marked as answer]
You can use PrusaSlicer which will work on Windows and my printer presets.
Lychee does also works in Linux.

About settings, File format (cxdlp) have that fields available but they aren't used by the printer, instead printer load settings from it menu and firmware which you can tune, that said change settings on UVtools in the file is useless.
Welcome to one of the worst file formats.

**Reply by rebeltaz** [reply]
Oh, ok... so it doesn't matter what I use in the settings? the printer overrides them anyway? I appreciate that. If I had known about the unusual file format, I might not have gotten this printer... good deal or no. Thanks again.

**Reply by sn4k3** [reply]
> Oh, ok... so it doesn't matter what I use in the settings? the printer overrides them anyway?

Correct. Printer override that. That's why you can't find such settings on thier slicer.

>  If I had known about the unusual file format, I might not have gotten this printer... good deal or no.

This chinese firmwares are all bad in something, can't understand either... They never care about customers, if you request something that will never see the light even if it's an fantastic ideia.

**Reply by rebeltaz** [reply]
I saw in the issues that there was an issue with the Halot Lite not recognizing files converted with uvtools (https://github.com/sn4k3/UVtools/issues/351). I am getting that same issue. I'm using v3.9.2 of uvtools. I have not updated the firmware on the printer, but I have tried both version 2 and 3 of the file format. Is there something else that I am missing or do I need to update the firmware? (I hate updating firmware... it tends to brick crap)..

**Reply by sn4k3** [reply]
That issue is not related, it was before UVtools had that in database.
Post your printer configurations here as screenshot from slicer you use

**Reply by rebeltaz** [reply]
I'm using chitubox

![Screenshot from 2022-12-17 12-09-53](https://user-images.githubusercontent.com/18725805/208255599-01b14fb5-5c51-4680-b86b-a62506a39a55.png)
![Screenshot from 2022-12-17 12-10-02](https://user-images.githubusercontent.com/18725805/208255603-ee4df5c7-17c4-4451-a656-76aa39f04998.png)
![Screenshot from 2022-12-17 12-10-11](https://user-images.githubusercontent.com/18725805/208255604-49d3229c-0700-49ff-95c4-18f3bc749ac6.png)
![Screenshot from 2022-12-17 12-10-20](https://user-images.githubusercontent.com/18725805/208255605-20bccc97-d18e-4ac4-a6fb-7d13f03891fb.png)

**Reply by sn4k3** [reply]
I can't see where's the problem. I tried to slice a file on chitubox which saves with cxdlp correctly and able to open in UVtools
![image](https://user-images.githubusercontent.com/113281/208256354-5534efb0-b986-477c-9051-6e5e4aa3aeea.png)

**Reply by rebeltaz** [reply]
Oh, I'm sorry. it's not uvtools that gives the error. it's the Halot Lite printer when I try to print a cxdlp file converted by uvtools

**Reply by sn4k3** [reply]
But why are you trying to convert file in UVtools? You just need to open the cxdlp do the modifications you want and click `Save`.

**Reply by rebeltaz** [reply]
Well, like I said I cannot run the slicer (Halot Box) because I run linux and it is not stable under wine. I am having to slice in chitubox and the Halot Lite doesn't accept the ctb format chitubox saves in. Are you saying that you are able to save in cxdlp format directly form chitubox? I don't see that as an option.

**Reply by sn4k3** [reply]
> Are you saying that you are able to save in cxdlp format directly form chitubox?

Yes, it shows to me, in fact i cant select ctb with that printer, only cxdlp.
Make sure you are in latest version

**Reply by rebeltaz** [reply]
Let me go download the newest version (I couldn't use the newest chitubox with  my mars) I appreciate it!

**Reply by sn4k3** [reply]
> I couldn't use the newest chitubox with my mars

You may need to update printer firmware to be compatible with CTBv4

**Reply by rebeltaz** [reply]
I'm sorry... I hate to keep bothering you. I downloaded chitubox v1.9.4. The only Creatlity printers I see are these. I selected the LD-006 which is the closest to my Halot Lite and even slicing it using that profile, it still only lets me select ctb format. 
![Screenshot from 2022-12-17 14-15-23](https://user-images.githubusercontent.com/18725805/208264581-5addef49-3159-444e-9843-2b87bcefee11.png)

**Reply by sn4k3** [reply]
![image](https://user-images.githubusercontent.com/113281/208265358-06ad8f89-c266-489e-b3e1-e080a02ad215.png)

I'm using [Chitubox Basic 1.9.5 Beta3.exe](https://drive.google.com/file/d/1h8ZHviPenh93UhWBxUb4FtigVHh_eHOS/view?usp=sharing) however that's for Windows and can't find for Linux. They may release the stable soon.
In meanwhile i think you should use PrusaSlicer with UVtools which have the printer already configured or Lychee both are free

**Reply by rebeltaz** [reply]
I'll try Prusa. I don't like the cloud computing idea behind lychee. Thank you, again, so much for all your help.

**Reply by sn4k3** [reply]
Follow this: https://github.com/sn4k3/UVtools/wiki/Setup-PrusaSlicer

**Reply by rebeltaz** [reply]
OK.. I tried that. I even updated the printer to the newest firmware. It is STILL telling me that "This Model does not match the printer!" whether I use version 2 or version 3 of the conversion. Which is fine because I **_REALLY_** didn't like PrusaSlicer at all! sigh.... I don't know. Right now I have a very nice looking paperweight. Once again... thank you for your help.

**Reply by sn4k3** [reply]
That's strange, can you update that file which says that error?

**Reply by sn4k3** [reply]
Ok I found why it didn't work. They have two machines with same resolution and UVtools get the first which is the CL-89, but yours it's CL-89L raising that problem.
To fix that after converting you can go to File -> Terminal, and send this two commands:

```
SlicerFile.MachineName = "CL-89L";
SlicerFile.Save();
```

![image](https://user-images.githubusercontent.com/113281/208269464-225bab89-6c89-4b2b-8d68-76712eefa43f.png)

Then it will work.

As an alternative in PrusaSlicer you can also clone my printer and simply call it "CL-89L" it will work too.

**Reply by rebeltaz** [reply]
Dude! That worked! Thank you SO much for all of your help! You have made me one very happy guy! 

Merry Christmas to you and yours! 

How can I send you a PayPal tip?

**Reply by sn4k3** [reply]
Glad it solved your problem :)

> Merry Christmas to you and yours!

Thanks, same to you 🎄

> How can I send you a PayPal tip?

UVtools -> Help -> Donate or Sponsor

**Comment by rebeltaz** [comment]
I'm just wondering if the newer version would still require me to do this for the Halot Lite

>Ok I found why it didn't work. They have two machines with same resolution and UVtools get the first which is the CL-89, but yours it's CL-89L raising that problem.

Just wondering whether upgrading would help me out or not. Thanks. and don't mean to bother you.

**Reply by sn4k3** [reply]
You mean upgrading UVtools? If you use PrusaSlicer with my profiles then yes. You should always update it.

**Reply by rebeltaz** [reply]
OH, no... With the version of UVTools I am running (3.12.2), when I convert a model from ctb to cxdlp fr use on the Halot Lite, I have to run 

```
 SlicerFile.MachineName = "CL-89L";
SlicerFile.Save();
```
in Terminal to correct it. I am wondering if the newer version takes care of that or if I would still need to run those commands. If I do... the version I've got's good.

**Reply by sn4k3** [reply]
With CTB to cxdlp there's no other way than that.
